From 5d411a25adadcd81c5d831a6185ae0d59ffd5a4e Mon Sep 17 00:00:00 2001
From: David Greaves <david@dgreaves.com>
Date: Mon, 25 Jun 2012 23:57:04 +0100
Subject: [PATCH 01/19] rpm-4.5.90-pkgconfig-path

---
 macros.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/macros.in b/macros.in
index 3a5bbcd..2e67e64 100644
--- a/macros.in
+++ b/macros.in
@@ -690,6 +690,8 @@ print (t)\
   export RPM_BUILD_ROOT}\
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
   export CLASSPATH}\
+  PKG_CONFIG_PATH=\"%{_libdir}/pkgconfig:%{_datadir}/pkgconfig\"\
+  export PKG_CONFIG_PATH\
   \
   %{verbose:set -x}%{!verbose:exec > /dev/null}\
   umask 022\
-- 
1.7.9

