From 1909dcd217cd543fa80aee55854fac3200573f93 Mon Sep 17 00:00:00 2001
From: David Greaves <david@dgreaves.com>
Date: Wed, 17 Apr 2013 20:41:08 +0000
Subject: [PATCH] rpm-macros

---
 configure.ac    |    1 +
 installplatform |    3 +++
 platform.in     |   20 ++++++++++++++++++++
 3 files changed, 24 insertions(+)

diff --git a/configure.ac b/configure.ac
index 45af439..91fa3b1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -830,6 +830,7 @@ unknown|pc|ibm|redhat|pld|mandrake|conectiva|lvr|yellowdog|caos|suse)
 	test -f /etc/redhat-release &&		RPMCANONVENDOR=redhat
 	test -f /etc/SuSE-release &&		RPMCANONVENDOR=suse
 	test -f /etc/pld-release &&		RPMCANONVENDOR=pld
+	test -f /etc/meego-release &&		RPMCANONVENDOR=meego
 	test -f /etc/mandrake-release &&	RPMCANONVENDOR=mandrake
 	test -f /etc/conectiva-release &&	RPMCANONVENDOR=conectiva
 	test -f /etc/lvr-release &&		RPMCANONVENDOR=lvr
diff --git a/installplatform b/installplatform
index 5973a18..09952f0 100755
--- a/installplatform
+++ b/installplatform
@@ -126,6 +126,9 @@ for SUBST in $SUBSTS ; do
   redhat)
     VENDORSED='-e s,^@redhat@,,'
     ;;
+  meego)
+    VENDORSED='-e s,^@meego@,,'
+    ;;
   apple)
     VENDORSED='-e s,^@apple@,,'
     ;;
diff --git a/platform.in b/platform.in
index a04d8fc..1b8eeb8 100644
--- a/platform.in
+++ b/platform.in
@@ -76,6 +76,26 @@
 @redhat@#%defattr(-,root,root,-)\
 @redhat@#%{nil}
 
+@meego@#---------------------------------------------------------------------
+@meego@#	Expanded at end of %prep
+@meego@#
+@meego@%__id_u		@__ID_U@
+@meego@%__chown_Rhf	@__CHOWN_RHF@
+@meego@%__chgrp_Rhf	@__CHGRP_RHF@
+@meego@%_fixperms	%{__chmod} -Rf @FIXPERMS@
+@meego@%__os_install_post    \
+@meego@    %{_rpmconfigdir}/brp-compress \
+@meego@    %{_rpmconfigdir}/brp-strip %{__strip} \
+@meego@    %{_rpmconfigdir}/brp-strip-static-archive %{__strip} \
+@meego@    %{_rpmconfigdir}/brp-strip-comment-note %{__strip} %{__objdump} \
+@meego@    %{_rpmconfigdir}/meego/find-docs.sh %{buildroot}\
+@meego@%{nil}
+@meego@#---------------------------------------------------------------------
+@meego@#	Always use %defattr(-,root,root) in %files (added in rpm-4.0.4)
+@meego@#
+@meego@#%files(n:f:)	%%files%{?-f: -f %{-f*}}%{?-n: -n %{-n*}} %{?1}\
+@meego@#%defattr(-,root,root,-)\
+@meego@#%{nil}
 
 @apple@#---------------------------------------------------------------------
 @apple@#	Apple Darwin vendor specific macros
-- 
1.7.10.3

