From 8e2d3e44b29c381a647f98100e5f9d6b2b51c651 Mon Sep 17 00:00:00 2001
From: David Greaves <david@dgreaves.com>
Date: Mon, 25 Jun 2012 23:57:04 +0100
Subject: [PATCH] rpm-4.9.0-nofsync

---
 macros.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index e8b359b..c634f3b 100644
--- a/macros.in
+++ b/macros.in
@@ -537,7 +537,7 @@ print (t)\
 # Misc BDB tuning options
 %__dbi_other			mp_mmapsize=128Mb mp_size=64Mb
 
-%_dbi_config			%{?__dbi_other}
+%_dbi_config			%{?__dbi_other} nofsync
 
 # "Packages" should have shared/exclusive fcntl(2) lock using "lockdbfd".
 %_dbi_config_Packages		%{?_dbi_config} lockdbfd
-- 
1.7.10.3

