From 20a3d8f266332d22dd8d31cfb1eeb1d248528a8a Mon Sep 17 00:00:00 2001
From: David Greaves <david@dgreaves.com>
Date: Mon, 25 Jun 2012 23:57:04 +0100
Subject: [PATCH] rpm-4.9.x-elfattr

---
 fileattrs/elf.attr |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fileattrs/elf.attr b/fileattrs/elf.attr
index bc6ce83..595b33e 100644
--- a/fileattrs/elf.attr
+++ b/fileattrs/elf.attr
@@ -1,4 +1,4 @@
 %__elf_provides		%{_rpmconfigdir}/elfdeps --provides %{?__filter_GLIBC_PRIVATE:--filter-private}
 %__elf_requires		%{_rpmconfigdir}/elfdeps --requires %{?__filter_GLIBC_PRIVATE:--filter-private}
-%__elf_magic		^(sticky )?ELF (32|64)-bit.*$
+%__elf_magic		^(setuid )?(setgid )?(sticky )?ELF (32|64)-bit.*$
 %__elf_flags		exeonly
-- 
1.7.10

