--- a/installplatform
+++ b/installplatform
@@ -46,10 +46,11 @@
 
   ARCH_INSTALL_POST='%{nil}'
   case "${ARCH}-${OS}" in
-    sparc64*-linux) LIB=lib64 ;;
-    s390x-linux) LIB=lib64 ;;
-    ppc64-linux|powerpc64-linux) LIB=lib64 ;;
-    x86_64-linux|amd64-linux|ia32e-linux) LIB=lib64 ;;
+    # Mer doesn't support multilib
+    #sparc64*-linux) LIB=lib64 ;;
+    #s390x-linux) LIB=lib64 ;;
+    #ppc64-linux|powerpc64-linux) LIB=lib64 ;;
+    #x86_64-linux|amd64-linux|ia32e-linux) LIB=lib64 ;;
     mips*-linux) LIB=lib ;;
     *) LIB=lib;;
   esac
