Index: rpm-4.9.0/scripts/find-debuginfo.sh
===================================================================
--- rpm-4.9.0.orig/scripts/find-debuginfo.sh
+++ rpm-4.9.0/scripts/find-debuginfo.sh
@@ -263,10 +263,17 @@ if [ -d "${RPM_BUILD_ROOT}/usr/lib" -o -
 
   (cd "${RPM_BUILD_ROOT}/usr"
    test ! -d lib/debug || find lib/debug ! -type d
-   test ! -d src/debug || find src/debug -mindepth 1 -maxdepth 1
   ) | sed 's,^,/usr/,' >> "$LISTFILE"
 fi
 
+: > "$SOURCEFILE"
+if [ -d "${RPM_BUILD_ROOT}/usr/src" ]; then
+  (cd "${RPM_BUILD_ROOT}/usr"
+   test ! -d src/debug || find src/debug -mindepth 1 -maxdepth 1
+  ) | sed 's,^,/usr/,' >> "$SOURCEFILE"
+fi
+
+
 # Append to $1 only the lines from stdin not already in the file.
 append_uniq()
 {
