--- rpm-4.8.0/scripts/perl.req~	2009-12-07 09:36:49.000000000 -0500
+++ rpm-4.8.0/scripts/perl.req	2010-03-26 09:20:54.513738528 -0400
@@ -209,16 +209,9 @@
       # if module is a number then both require and use interpret that
       # to mean that a particular version of perl is specified
 
-      my $ver=$1;
-      if ($ver =~ /5.00/) {
-        print "perl >= 0:$ver\n";
+        my $ver=$1;
+        print "perl\n";
         next;
-      }
-      else {
-        print "perl >= 1:$ver\n";
-        next;
-      }
-
       };
 
       # ph files do not use the package name inside the file.
